packages/halo/src/react/with-default-theme.js

```diff
@@ -0,0 +1,28 @@
+import hoistNonReactStatics from 'hoist-non-react-statics'
+import PropTypes from 'prop-types'
+import React from 'react'
+
+import { defaultName as themeDefaultName } from '@pluralsight/ps-design-system-theme/react'
+
+const getDisplayName = Component => {
+  if (typeof Component === 'string') return Component
+
+  if (!Component) return undefined
+
+  return Component.displayName || Component.name || 'Component'
+}
+
+export default function withDefaultTheme (BaseComponent) {
```

Comment:
I don't love this name - to me it might mean the component is explicitly using the `default` theme. what could be a better name, `withThemeDefault`? `withFallbackTheme`?